### PR TITLE
Update ca_renewables.star

### PR DESCRIPTION
**Pull request by burritojustice**
Hi!

Fixed some bugs and made some enhancements to the California Renewables app:

- no longer adding negative power sources (charging batteries, exports) to total power in order to get accurate renewable percentages (this was causing total renewable power to exceed 100% of supply and was breaking the chart)
- changed default animation delay, added dropdown for additional timing options (previous delay was too long for app cycle and wasn't animating at all)
- added config toggles to display and hide power sources
- hid geothermal, biogas, biomass, small hydro power sources by default (all very small)
- added batteries (solar/wind charged, negative while charging, often largest power source at night)
- added option to show batteries charging below the x-axis
- added large hydro (not an official renewable)
- note that toggling off batteries and large hydro will remove them from the renewable total, while toggling off other official renewable totals (biogas, etc) will keep them in the total)
- modified chart colors for better visibility

### CODE REVIEW

**Comment by burritojustice** [comment]
I have read the CLA Document and I hereby sign the CLA

File: apps/carenewables/ca_renewables.star [hunk 1]
```diff
@@ -117,15 +171,109 @@ def main():
...
+                name = "Geothermal",
+                desc = "Display geothermal power",
+                icon = "mugHot",
+                default = False,
+            ),  
+            schema.Toggle(
+                id = "biogas",
+                name = "Biogas",
+                desc = "Display biogas power",
+                icon = "fireDlameSimple",
```
Comment (rohansingh) [inline apps/carenewables/ca_renewables.star:254]:
```suggestion
                icon = "fireFlameSimple",
```

**Comment by burritojustice** [comment]
hey @rohansingh, @matslina -- let me know if there's anything else I need to do. Wanted to emphasize that it's possible to use the config settings to recreate the previous interface if someone doesn't want to see batteries and hydro in the cycle.

**Comment by matslina** [comment]
@sloanesturz Did you see this? Looks ok to me but LMK if you have any thoughts before we merge.

**Review by sloanesturz — COMMENTED** [review]
Really cool! I've noticed the breaking chart too but haven't had time to look at why. 

Thanks for the fixes and new features.

**Review by sloanesturz — COMMENTED** [review]
Really cool! I've noticed the breaking chart too but haven't had time to look at why. 

Thanks for the fixes and new features.

**Comment by sloanesturz** [comment]
@burritojustice if you have any other ideas for energy and climate Tidbyt apps, let me know!

**Comment by burritojustice** [comment]
ha, funny you should mention that -- not sure if you have seen gridstatus.io but they have great data from multiple ISOs, and they have an API.Thinking about some sort of California vs Texas energy comparison...

https://www.gridstatus.io
https://api.gridstatus.io/docs#section/Getting-Started

**Pull request merged**